spec/services/measurements_creator_spec.rb

```diff
@@ -0,0 +1,148 @@
+require "spec_helper"
+
+describe MeasurementsCreator do
+  describe "#self.call" do
+    before() do
+      @arrival_utc_time = Time.utc(2018, 11, 19)
+      allow(Time).to receive(:current).and_return(@arrival_utc_time)
+    end
+
+    context "when there is only one measurement" do
+      it "creates a measurement" do
+        session = create_session!
+        stream = create_stream!(session: session)
+
+        MeasurementsCreator.call(stream, single_measurement_attributes)
+
+        expect(Measurement.count).to eq(1)
+      end
+
+      it "when sessions is fixed creates a measurement with utc time" do
+        session = create_session!(type: "FixedSession")
+        stream = create_stream!(session: session)
+
+        MeasurementsCreator.call(stream, single_measurement_attributes)
+
+        expect(Measurement.first.arrival_utc_time).to eq(@arrival_utc_time)
+      end
+
+      it "when session is moblie creates a measurement without utc time" do
+        session = create_session!(type: "MoblieSession")
+        stream = create_stream!(session: session)
+
+        MeasurementsCreator.call(stream, single_measurement_attributes)
+
+        expect(Measurement.first.arrival_utc_time).to be_nil
+      end
+    end
+
+    context "when there are more than one measurements" do
+      it "schedules the creation of measurements" do
+        session = create_session!
+        stream = create_stream!(session: session)
+
+        MeasurementsCreator.call(stream, multiple_measurements_attributes)
+
+        assert_equal 1, AsyncMeasurementsCreator.jobs.size
+      end
+    end
+  end
+
+  describe "#call" do
+    let(:streams_repository) { double(StreamsRepository.new) }
+    let(:measurements_creator) { MeasurementsCreator.new(streams_repository) }
+    let(:stream) { double(Stream.new) }
```

Comment:
Didn't know you can pass an object inside `double`. What's the advantage over just `double`?

Reply:
Sorry, it's a mistake. I didn't know the exact syntax and I found a similar example somewhere, cause I couldn't find the documentation for RSpec 2.99. 
Anyway, now I read the documentation 3.8 and it turns out the first argument passed to double is just its name, that's displayed if the test fails. I checked and it works the same in our rspec, so I will change this to `double("stream")`: 
![screen shot 2018-11-20 at 11 46 04](https://user-images.githubusercontent.com/30773137/48768930-ac161280-ecba-11e8-874b-3a568c441161.png)
![screen shot 2018-11-20 at 11 46 33](https://user-images.githubusercontent.com/30773137/48768935-ad473f80-ecba-11e8-8a6e-eb325d3d2c39.png)